### DISCUSSION TITLE
How to test button with image

### DISCUSSION THREAD

**Discussion by ckunchur** [CS342/.github#19] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Xcode, Simulator & Previews

### Description

I cannot find a way to write a test for a button that has an image rather than a text label. Any suggestions?

### Reproduction

N/A

### Expected behavior

N/A

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by vishnuravi** [comment]
Hey @ckunchur, if the button only has an image, you can use the name of the image to test it - see the example here: https://smashswift.com/how-to-test-buttons-in-ui-tests-in-swiftui/

**Reply by ckunchur** [reply]
I did previously try that using the image filename, but the test doesn't pass (passing in the image name only seems to work if using a built in system icon).

I just tried adding an accessibilityIdentifier as the article you shared mentions, but I don't seem to be wrapping the button correctly, as I get several errors. 
```
Button( {
                openURL(instaURL)
            }
        label: {
            Bundle.module.image(fromFileNamed: "Instagram")
                .resizable()
                .padding(.horizontal, 5)
                .frame(width: 65, height: 40)
            }).accessibilityIdentifier("Instagram")
```

**Comment by PSchmiedmayer** [comment, marked as answer]
Hi @ckunchur,

As @vishnuravi pointed out, the accessibility modifier is the right way to go, you are on a good path!
Your code example that you have shared only as a syntax error and you have to attach the accessibility modifier to the complete button:
```swift
Button(
    action: {
        // ... your action
    },
    label: {
        Bundle.module.image(fromFileNamed: "Instagram")
            .resizable()
            .padding(.horizontal, 5)
            .frame(width: 65, height: 40)
    }
)
    .accessibilityIdentifier("Instagram")
```

You can then tap the button in your UI test using the following call:
```swift
app.buttons["Instagram"].tap()
```

**Reply by PSchmiedmayer** [reply]
@ckunchur Please let us know if this solved your issue! 👍
